pkg/migration/migration.go

```diff
@@ -0,0 +1,145 @@
+	logger.DefaultLog("Create Kubernetes Client")
+	client, err := k8sutil.NewClient(kubeConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create kubernetes client: %v", err)
+	}
+
+	logger.DefaultLog("List all the PVC from the source storageclass")
+	pvcs, err := k8sutil.ListAllPVCWithStorageclass(client, sourceStorageClass)
+	if err != nil {
+		return fmt.Errorf("failed to list PVCs from the storageclass: %v", err)
+	}
+	if pvcs == nil || len(*pvcs) == 0 {
+		logger.DefaultLog("no PVCs found with storageclass: %v", sourceStorageClass)
+		return nil
+	}
+	logger.DefaultLog("%d PVCs found with sourceStorageClass %s ", len(*pvcs), sourceStorageClass)
+
+	logger.DefaultLog("Start Migration of PVCs to CSI")
+	for _, pvc := range *pvcs {
+		err = migratePVC(client, pvc, destinationStorageClass, rookNamespace, cephClusterNamespace)
+		if err != nil {
+			return fmt.Errorf("failed to migrate PVC %s : %v", pvc.Name, err)
+		}
+	}
+	logger.DefaultLog("Successfully migrated all the PVCs to CSI")
+
+	return err
+}
+
+// migratePVC migrates a PVC to CSI.
+func migratePVC(client *k8s.Clientset, pvc v1.PersistentVolumeClaim, destinationStorageClass,
+	rookNamespace, cephClusterNamespace string) error {
+
+	logger.DefaultLog("migrating PVC %q", pvc.Name)
+
+	logger.DefaultLog("Fetch PV information from PVC %s", pvc.Name)
+	pv, err := k8sutil.GetPV(client, pvc.Spec.VolumeName)
+	if err != nil {
+		return fmt.Errorf("failed to get PV object with name %s: %v", pvc.Spec.VolumeName, err)
+	}
+	logger.DefaultLog("PV found %q ", pv.Name)
+
+	logger.DefaultLog("Update Reclaim policy from Delete to Reclaim for PV: %s", pv.Name)
+	err = k8sutil.UpdateReclaimPolicy(client, pv)
+	if err != nil {
+		return fmt.Errorf("failed to update ReclaimPolicy for PV object %s: %v", pvc.Spec.VolumeName, err)
+	}
+
+	logger.DefaultLog("Retrieving old ceph volume name from PV object: %s", pv.Name)
+	rbdImageName := k8sutil.GetVolumeName(pv)
+	if rbdImageName == "" {
+		return fmt.Errorf("rbdImageName cannot be empty in Flex PV object: %v", pv)
+	}
+	logger.DefaultLog("Flex rbdImageName name %v ", rbdImageName)
+
+	logger.DefaultLog("Deleting pvc object: %s", pvc.Name)
+	err = k8sutil.DeletePVC(client, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+	if err != nil {
+		return fmt.Errorf("failed to Delete PVC object %s: %v", pvc.Name, err)
+	}
+
+	logger.DefaultLog("Generate new PVC with same name in destination storageclass")
+	csiPVC := k8sutil.GenerateCSIPVC(destinationStorageClass, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+	logger.DefaultLog("Structure of the generated PVC in destination Storageclass %v ", csiPVC)
+
+	logger.DefaultLog("Create new csi pvc")
+	csiPV, err := k8sutil.CreatePVC(client, csiPVC, pvcCreateTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to Create CSI PVC object %s: %v", pvc.Name, err)
+	}
+	logger.DefaultLog("New PVC with same name created in CSI. Persistent Volume: %v ", csiPV)
```

Comment:
Logging the name should be sufficient, otherwise it is too verbose
```suggestion
	logger.DefaultLog("New PVC with same name %q created", csiPVC.Name)
```